lib/commands/cloud-codesign.ts

```diff
@@ -0,0 +1,34 @@
+export class CloudCodesign implements ICommand {
+	public allowedParameters: ICommandParameter[];
+
+	constructor(private $logger: ILogger,
+		private $projectData: IProjectData,
+		private $prompter: IPrompter,
+		private $options: IOptions,
+		private $cloudCodesignService: ICloudCodesignService) {
+		this.$projectData.initializeProjectData();
+	}
+
+	public async execute(args: string[]): Promise<void> {
+		let username = args[0];
+		let password = args[1];
+
+		if (!username) {
+			username = await this.$prompter.getString("Apple ID", { allowEmpty: false });
```

Comment:
We had a discussion about the parameters approach with @Mitko-Kerezov and @TsvetanMilanov . The idea here is to have at least some prompts as you do not have explicit --options and you do not have help. This way you do not have to know the exact order of the parameters and all of them as a matter of fact.  Similar implementation may be found here - https://github.com/NativeScript/nativescript-cli/blob/master/lib/commands/appstore-upload.ts#L38 , so users are familiar with this behavior.

I know this is also meant to be used for automation scenarios, but in such cases, I think it is highly possible the parameters are passed correctly. If not it will be one time debug to find that you are missing parameters. 
Of course, I am open to further discussions, may be @Mitko-Kerezov and @TsvetanMilanov  might share opinion?

Reply:
`I know this is also meant to be used for automation scenarios, but in such cases, I think it is highly possible the parameters are passed correctly.`

**highly possible** does not mean that they are passed correctly. 
`it will be one time debug to find that you are missing parameters.` - do you expect users to debug CLI ?

The idea is to use `isInteractive()` method and in case not and in case all required arguments are not passed, to fail with descriptive message instead of hanging.
You can handle this in `canExecute` method of the command.